test/connection_server.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+const child_process = require('child_process');
+const mkdirp = require('mkdirp');
+const rimraf = require('rimraf');
+
+// For starting/stopping mongod to test connections
+// Adapted from mongodb-topology-manager:
+// https://github.com/mongodb-js/mongodb-topology-manager/blob/master/lib/server.js
+class Server {
+  constructor(binary, options) {
+    this.binary = binary;
+    this.port = options['port'] || 27017;
+    this.bind_ip = options['bind_ip'] || '127.0.0.1';
+    this.dbpath = options['dbpath'] || './data';
+  }
+
+  // Start server
+  start() {
+    const self = this;
+    return new Promise(function(resolve, reject) {
+      self.purge().then(function() {
+        // Spawn server process
+        self.server = child_process.spawn(self.binary,
+          ['--port', self.port, '--bind_ip',
+            self.bind_ip, '--dbpath', self.dbpath]);
+
+        // Wait for ready
+        self.server.stdout.on('data', function(data) {
+          if (data.toString().includes('Waiting for connections')) {
```

Comment:
This change means that this logic won't work with MongoDB < 4.4, right?

Reply:
I'll look into it, and try to improve compatibility with older versions.

Reply:
I've reverted `test.yml`, so it should be back to testing 4.0.2.